### PR TITLE
feat(Health): roll transactionAnchorRetryInfo into health collection

### PR DESCRIPTION
**Pull request by krobi64**
[PR Process](https://github.com/poetapp/documentation/blob/master/process/pr-review.md#pr-process) - [PR Review Checklist](https://github.com/poetapp/documentation/blob/master/process/pr-review.md#pr-review-checklist)

### Release
 Semantic release is enabled for this repository. Make sure you follow the right commit message convention. 
 We're using semantic-release's default — [Angular Commit Message Conventions](https://github.com/angular/angular.js/blob/master/DEVELOPERS.md#-git-commit-guidelines).

## Description of Changes

References #832 

On getHealth event,
1. get the grouped data from the IPFSDirectoryHash collection
1. store it in the Health collection under `name: 'transactionAnchorRetryInfo'`

### CODE REVIEW

File: src/Health/HealthController.ts [hunk 1]
```diff
@@ -134,6 +150,13 @@ export class HealthController {
     return ipfsInfo
   }
 
+  private async updateTransactionAnchorRetryInfo(
+    transactionAnchorRetryInfo: TransactionAnchorRetryInfo,
+  ): Promise<object> {
```
Comment (kennylavender) [inline src/Health/HealthController.ts]:
```suggestion
  ): Promise<TransactionAnchorRetryInfo> {
```

File: src/Health/HealthController.ts [hunk 2]
```diff
@@ -40,6 +45,7 @@ export const isFailureHard = (failureType: string) => failureType === 'HARD'
 
 export interface Dependencies {
   readonly healthDAO: HealthDAO
+  readonly ipfsDirectoryHasDAO: IPFSDirectoryHashDAO
```
Comment (lautarodragan) [inline src/Health/HealthController.ts]: typo I think. HasDAO -> HashDAO? @krobi64

Comment (krobi64) [reply src/Health/HealthController.ts]: 🤦‍♀️

Comment (krobi64) [reply src/Health/HealthController.ts]: I'll fix in the next PR./

**Pull request merged**